easypy/caching.py

```diff
@@ -171,29 +172,73 @@ def clear():
 
     return deco
 
+if sys.version_info < (3, 5):
+    def _apply_defaults(bound_arguments):
+        """
+        Set default values for missing arguments. (from Python3.5)
+        """
+        from collections import OrderedDict
+        from inspect import _empty, _VAR_POSITIONAL, _VAR_KEYWORD
+
+        arguments = bound_arguments.arguments
+        new_arguments = []
+        for name, param in bound_arguments._signature.parameters.items():
+            try:
+                new_arguments.append((name, arguments[name]))
+            except KeyError:
+                if param.default is not _empty:
+                    val = param.default
+                elif param.kind is _VAR_POSITIONAL:
+                    val = ()
+                elif param.kind is _VAR_KEYWORD:
+                    val = {}
+                else:
+                    # This BoundArguments was likely produced by
+                    # Signature.bind_partial().
+                    continue
+                new_arguments.append((name, val))
+        bound_arguments.arguments = OrderedDict(new_arguments)
```

Comment:
Why convert a list of pairs to an `OrderedDict`? Why not just make an empty `OrderedDict` and populate it?

Reply:
I copied the code from python3.5 sources, didn't want to change anything to official code.